strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/artifact/coordinates/GenericArtifactCoordinates.java

```diff
@@ -0,0 +1,24 @@
+package org.carlspring.strongbox.artifact.coordinates;
+
+import java.util.Map;
+
+import org.carlspring.strongbox.data.domain.DomainObject;
+import org.carlspring.strongbox.data.domain.EntityHierarchyNode;
+
+/**
+ * @author sbespalov
+ *
+ */
+public interface GenericArtifactCoordinates extends DomainObject, EntityHierarchyNode<GenericArtifactCoordinates>
+{
+
```

Comment:
Shouldn't this interface also expect and `id` for the artifact? If I recall correctly, the most basic coordinates that I had defined (a very long time ago, of course), were `id` and `version`. Have things changed since then...?

Reply:
the `id` is in the `ArtifactCoordinates` interface now. From the top point of view nothing changed because it's assumed that you will not use `GenericArtifactCoordinates` directly, it needed just to store coordinates properly in DB according the entities hierarchy.